### PR TITLE
c++14 multipole for Intel

### PR DESCRIPTION
**Pull request by loriab**
## Description
Get Psi4 running correctly with Intel (2021.4) again. I'm sorry to uglify the nice structured bindings (thanks for the name, @robertodr). Edit with any better suggestions. I've got full tests running locally.

## Todos

- [ ] Feature1
- [ ] Feature2

## Questions
- [ ] Question1

## Checklist
- [ ] Tests added for any new features
- [ ] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [ ] Ready for review
- [ ] Ready for merge

### CODE REVIEW

**Review by JonathonMisiewicz — APPROVED** [review]

**Review by jturney — APPROVED** [review]

File: psi4/src/psi4/libmints/multipoles.cc [hunk 1]
```diff
@@ -186,10 +186,26 @@ void MultipoleInt::compute_pair(const libint2::Shell& s1, const libint2::Shell&
...
+                    const auto ez = dummy1[2];
+                    const auto index0 = dummy1[3];
                     ao12 = 0;
-                    for (const auto& [l1, m1, n1, index1] : comps_am1) {
-                        for (const auto& [l2, m2, n2, index2] : comps_am2) {
+                    for (const auto& dummy2 : comps_am1) {
+                        const auto l1 = dummy2[0];
+                        const auto m1 = dummy2[1];
+                        const auto n1 = dummy2[2];
+                        const auto index1 = dummy2[3];
```
Comment (maxscheurer) [inline psi4/src/psi4/libmints/multipoles.cc:203]:
```suggestion
                    for (const auto& comp_am1 : comps_am1) {
                        const auto l1 = comp_am1[0];
                        const auto m1 = comp_am1[1];
                        const auto n1 = comp_am1[2];
                        const auto index1 = comp_am1[3];
```

**Review by maxscheurer — COMMENTED** [review]
Why is this not needed for `angularmomentum.cc`, where structured bindings are used, too?

**Comment by JonathonMisiewicz** [comment]
> Why is this not needed for `angularmomentum.cc`, where structured bindings are used, too?

See [comment](https://github.com/psi4/psi4/blob/b644db4e34f9e5f134772cc62e3fd012c84ebcfc/psi4/src/psi4/libmints/CMakeLists.txt#L55-L58).

**Comment by maxscheurer** [comment]
Well, this was my bad... I accidentally included `engine.h` in `multipoles.cc`. I'm really sorry... :cry:

**Comment by JonathonMisiewicz** [comment]
> Well, this was my bad... I accidentally included `engine.h` in `multipoles.cc`. I'm really sorry... 😢

As in, we can just remove the include instead?

**Comment by maxscheurer** [comment]
> As in, we can just remove the include instead?

Yes, I sorted it out with @loriab on Slack. The only required change (I think) is to replace the engine include with shell.h include.

**Review by maxscheurer — APPROVED** [review]

**Pull request merged**